localstack/services/dynamodb/provider.py

```diff
@@ -198,104 +230,88 @@ def shutdown(self):
         self.executor.shutdown(wait=False)
 
     def forward_to_targets(
-        self, account_id: str, region_name: str, records: List[Dict], background: bool = True
+        self, account_id: str, region_name: str, records_map: RecordsMap, background: bool = True
     ):
         if background:
             self.executor.submit(
                 self._forward,
                 account_id=account_id,
                 region_name=region_name,
-                records=records,
+                records_map=records_map,
             )
         else:
-            self._forward(account_id, region_name, records)
+            self._forward(account_id, region_name, records_map)
 
-    def _forward(self, account_id: str, region_name: str, records: List[Dict]):
-        # forward to kinesis stream
-        records_to_kinesis = copy.deepcopy(records)
-        self.forward_to_kinesis_stream(records_to_kinesis)
+    def _forward(self, account_id: str, region_name: str, records_map: RecordsMap):
+        # You can only stream data from DynamoDB to Kinesis Data Streams in the same AWS account and AWS Region as your
+        # table.
+        # You can only stream data from a DynamoDB table to one Kinesis data stream.
+        try:
+            self.forward_to_kinesis_stream(account_id, region_name, records_map)
+        except Exception:
+            LOG.exception("Error while publishing to DynamoDB streams")
 
-        # forward to ddb_streams
-        forward_records = self.prepare_records_to_forward_to_ddb_stream(records)
-        records_to_ddb = copy.deepcopy(forward_records)
-        self.forward_to_ddb_stream(account_id, region_name, records_to_ddb)
+        try:
+            self.forward_to_ddb_stream(account_id, region_name, records_map)
+        except Exception:
+            LOG.exception("Error while publishing to Kinesis streams")
```

Comment:
The error messages seem exchanged between this and the previous try-except blocks